Add docs about CSS build system

Aljullu opened the pull request:
One of the suggestions in #2818 was to add some docs about how the CSS build system works.

senadir commented on docs/contributors/css-build-system.md:
```diff
@@ -0,0 +1,25 @@
+-   `style.css`:
+    -   Loaded in the editor and the frontend.
+    -   Bundles files located in:
+        -   Inside other `/assets/` subdirectories like `/assets/css/`, `/assets/js/base/components/`...
```
This is confusing, does it mean it also includes `/assets/js/components/`?

senadir commented on docs/contributors/css-build-system.md:
```diff
@@ -0,0 +1,25 @@
+# CSS build system
```
It was confusing for me to find the relevant code that is responsible for this build in webpack, would it make sense to include such files and/or the PR that introduced them?

haszari commented on docs/contributors/README.md:
```diff
@@ -10,3 +10,4 @@ This folder contains documentation for developers and contributors looking to ge
 | [JavaScript Testing](javascript-testing.md)                      | This doc explains how to run automated JavaScript tests.                            |
 | [Component Testing & Development (with Storybook)](storybook.md) | This doc explains how to use Storybook to test and develop components in isolation. |
 | [Block Script Assets](block-assets.md)                           | This doc explains how Block Script Assets are loaded and used.                      |
+| [CSS build system](css-build-system.md)                          | This doc explains how CSS is build.                                                 |
```
Typo: `This doc explains how CSS is built.`

haszari commented on docs/contributors/css-build-system.md:
```diff
@@ -0,0 +1,25 @@
+
+## Legacy builds
+
+Legacy builds use their own CSS files with the suffix `-legacy`, this allows saving some bytes because non-legacy block styles are not included. However, all components' styles are included no matter if they are only used by non-legacy blocks. So file size difference is not that big. See [#2818](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2818) for more details.
```
Is the `However, all components' styles…` info essential to document here? Feels less important, more of a "by the way" or an appendix/footnote (or code comment).

Aljullu commented:
Thanks for all the good feedback! I updated the document with your suggestions. I removed the explanations on how CSS files are split and instead linked `webpack-configs.js`, hopefully this will make the docs simpler. I also added a list of `Relevant files` so it's easier to find where in the codebase one can find the config.
